library/alloc/src/rc.rs

```diff
@@ -386,7 +386,35 @@ impl<T> Rc<T> {
     /// returns, calling [`upgrade`] on the weak reference inside `data_fn` will
     /// fail and result in a `None` value.
     ///
+    /// # Why using this method?
```

Comment:
The documentation of every function is (should be?) answering this question. If we only get to this question after three paragraphs of documentation, it makes me think we should change/(re)move those and directly start with explaining the purpose of this function.

Reply:
I'll move it the explanation on top then.